app/helpers/mb_form_builder.rb

```diff
@@ -397,3 +399,39 @@ def check_box_with_label(label_text, method, options = {})
     HTML
   end
 end
+
+module ActionView
+  module Helpers
+    class DateTimeSelector
+
+      # Given an ordering of datetime components, create the selection HTML
+      # and join them with their appropriate separators.
+      def build_selects_from_types(order)
+        select = ""
+        order.reverse_each do |type|
+          separator = separator(type)
+          select.insert(0, separator.to_s + send("select_#{type}").to_s + "</div>")
+        end
+        select.html_safe
+      end
+
+      # Returns the separator for a given datetime component.
+      def separator(type)
+        return "" if @options[:use_hidden]
+        field_name = "#{@options[:prefix]}_#{@options[:field_name]}"
+        case type
+          when :year
+            "<div class='select'><label for='#{field_name}_1i' class='sr-only'>Year</label>"
+          when :month
+            "<div class='select'><label for='#{field_name}_2i' class='sr-only'>Month</label>"
+          when :day
+            "<div class='select'><label for='#{field_name}_3i' class='sr-only'>Day</label>"
+          when :hour
+            (@options[:discard_year] && @options[:discard_day]) ? "" : @options[:datetime_separator]
+          when :minute, :second
+            @options[:"discard_#{type}"] ? "" : @options[:time_separator]
+        end
+      end
+    end
+  end
+end
```

Comment:
Would it feel better if our `mb_date_select` just grabbed each select individually instead of monkey patching (`select_year`, `select_month`, `select_day`) (assuming it's possible to do so and get labels)?

Reply:
yeah we kinda looked into that and in the end it felt like it was going to take more time. *but* perhaps the time investment is worth it?

Reply:
I'm gonna explore the individual `select_*` methods approach today.